proxy.go

```diff
@@ -0,0 +1,292 @@
+package tunnel
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"sync"
+
+	"github.com/koding/logging"
+	"github.com/koding/tunnel/proto"
+)
+
+// Proxy is responsible for forwarding remote connection to local server and writing the response back.
+type Proxy interface {
```

Comment:
Sure we could do that, I thought it was cleaner with the interface. With function awkward naming can be avoided I like that.